core/src/com/ericc/the/game/systems/logic/FieldOfViewSystem.java

```diff
@@ -0,0 +1,204 @@
+package com.ericc.the.game.systems.logic;
+
+import com.badlogic.ashley.core.Engine;
+import com.badlogic.ashley.core.Entity;
+import com.badlogic.ashley.core.EntitySystem;
+import com.badlogic.ashley.core.Family;
+import com.badlogic.ashley.utils.ImmutableArray;
+import com.badlogic.gdx.math.GridPoint2;
+import com.badlogic.gdx.math.MathUtils;
+import com.ericc.the.game.Mappers;
+import com.ericc.the.game.components.FieldOfViewComponent;
+import com.ericc.the.game.components.PlayerComponent;
+import com.ericc.the.game.components.PositionComponent;
+import com.ericc.the.game.helpers.ScreenBoundsGetter;
+import com.ericc.the.game.map.Map;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class FieldOfViewSystem extends EntitySystem {
+
+    private Map map;
+    private ImmutableArray<Entity> otherMobs;
+    private ImmutableArray<Entity> heroes;
+    private ScreenBoundsGetter sbh;
+
+    // a helper data structure with possible moves from one tile in horizontal and vertical directions
+    private static ArrayList<GridPoint2> moves =
+            new ArrayList<>(Arrays.asList(
+                    new GridPoint2(1, 0),
+                    new GridPoint2(0, 1),
+                    new GridPoint2(-1, 0),
+                    new GridPoint2(0, -1)
+            ));
+
+    // stores moves on diagonals
+    private static ArrayList<GridPoint2> cornerMoves =
+            new ArrayList<>(Arrays.asList(
+                    new GridPoint2(1, 1),
+                    new GridPoint2(-1, -1),
+                    new GridPoint2(-1, 1),
+                    new GridPoint2(1, -1)
+            ));
+
+    private static ArrayList<ArrayList<GridPoint2>> corners =
+            new ArrayList<>(
```

Comment:
trochę nie rozumiem czemu robisz new ArrayList z Arrays.asList

Reply:
może być po prostu Arrays.asList i zmienić return type na List<List<GridPoint2>>. Jeśli masz obawy wydajnościowe (ale te listy i tak mają po 4 elementy, więc nawet iterowanie się po nich za każdym razem to jest żaden koszt...), to ta lista którą zwraca Arrays.asList jest prawie tym samym co ArrayList (prawdopodobnie jest *dokładnie* tym samym, ale trzeba by zajrzeć do implementacji javy :p) 

> public static <T> List<T> asList(T... a)
> Returns a fixed-size list backed by the specified array. (Changes to the returned list "write through" to the array.) This method acts as bridge between array-based and collection-based APIs, in combination with Collection.toArray(). The returned list is serializable and implements RandomAccess.